lib/book-services.js

```diff
@@ -227,12 +229,31 @@ class BookServices {
         Step 6: Create a custom cover
     */
 
-    static createCustomCover(book) {
-        if (!book.hasCustomCover()) {
+    static createCover(book) {
+        if (book.hasDefaultCover()) {
             tracker.setStatus(book.getId(), STATUS_TYPES.CREATING_COVER);
             return StylingService.writeOnCover(book, book.getTitle());
         }
-        return Promise.resolve(book);
+
+        // any non-default is a remote cover
+        const coverPath = book.getCoverPath();
+        const coverFileName = `${book.getId()}-cover.png`;
+        const downloadPath = path.join(Config.COVERS_TMP, coverFileName);
+
+        return request.get({
```

Comment:
I actually implemented a `ContentDownloader` class. Looking at it now it isn't exactly the most elegant abstraction, but it does give some benefits:

- Here you've hardcoded the filename to be `.png` - the ContentDownloader class actually parses the URL / Content Type and generates a filename with the correct type.
- You can set limits on the size of the download (if someone tries to trick the server into downloading a large image - it will cancel the request early). 
- It should hide away some amount of response parsing and simplify stuff...
- It uses asynchronous APIs (with how CPU heavy EpubPress is, becomes especially important to use the Async methods).

Believe it would look something like this:

```js
const contentDownloader = new ContentDownloader(coverPath);

return contentDownloader.download().then((result) => {
    book.setCoverPath(result.path);
    return book;
}).catch(log.exception('BookServices.convertSectionContent'));
```

Reply:
Awesome - will check it out and convert it.

Reply:
Converted in https://github.com/haroldtreen/epub-press/pull/39/commits/92222c4cf2d0a39bef3e17e289f8f245f8666276